### PR TITLE
KAA-1277: [C++ SDK] Update C++ build guide for Windows OS

### PR DESCRIPTION
**Pull request by RostakaGmfun**
/cc @vadimol

### CODE REVIEW

**Comment by Slash32** [comment]
LGTM

File: datacollectiondemo/source/cpp/build.bat [hunk 1]
```diff
@@ -76,13 +76,14 @@ goto help
         md %BUILD_DIR%
         cd %BUILD_DIR%
         cmake -G "NMake Makefiles" ^
-	      -DKAA_DEBUG_ENABLED=1 ^
+              -DCMAKE_BUILD_TYPE=Debug ^
               -DKAA_WITHOUT_EVENTS=1 ^
-              -DKAA_WITHOUT_CONFIGURATION=1 ^
+              -DKAA_WITHOUT_LOGGING=1 ^
```
Comment (vadimol) [inline datacollectiondemo/source/cpp/build.bat]: Should we disable Logging in the Data collection demo? AFAIK this is a key feature here. Am I wrong?

**Comment by vadimol** [comment]
See comment about data collection demo.

File: datacollectiondemo/source/cpp/build.bat (hunk 1 above)
Comment (RostakaGmfun) [inline datacollectiondemo/source/cpp/build.bat]: Thanks, seems like a copy-paste error. Will fix ASAP

**Comment by vadimol** [comment]
What is about:
./sparkdemo/source/cpp/build.bat
./vehicletelemetry/source/cpp/build.bat
./stormdemo/source/cpp/build.bat
? 
Should these demos be updated too?

**Comment by RostakaGmfun** [comment]
Fixed, thanks

**Comment by vadimol** [comment]
Looks good to me

**Comment by GeorgePostolakiy** [comment]
1. Wrong boost link (actual: boost x32 / expected: boost x64)
   Please change:
   `https://sourceforge.net/projects/boost/files/boost-binaries/1.60.0/boost_1_60_0-msvc-14.0-32.exe/download`
   to
   `https://sourceforge.net/projects/boost/files/boost-binaries/1.60.0/boost_1_60_0-msvc-14.0-64.exe/download`
2. Please add 7zip step "to Installing prerequisites" (do not forget to mention about PATH)
3. Please pay attention on env.bat => botan => PATH. -- need to add `%ROOT_PATH%\lib`

**Pull request merged**